sycl/unittests/queue/GetProfilingInfo.cpp

```diff
@@ -0,0 +1,182 @@
+    (void)end_time;
+  });
+}
+
+TEST(GetProfilingInfo, command_exception_check) {
+  cl::sycl::platform Plt{cl::sycl::default_selector{}};
+  if (Plt.is_host()) {
+    GTEST_SKIP();
+  }
+  sycl::unittest::PiMock Mock{Plt};
+  setupDefaultMockAPIs(Mock);
+  Mock.redefine<sycl::detail::PiApiKind::piEventGetProfilingInfo>(
+      redefinedPiEventGetProfilingInfo);
+
+  const sycl::device Dev = Plt.get_devices()[0];
+  static sycl::unittest::PiImage DevImage_1 =
+      generateTestImage<InfoTestKernel>();
+
+  static sycl::unittest::PiImageArray<1> DevImageArray = {&DevImage_1};
+  auto KernelID_1 = sycl::get_kernel_id<InfoTestKernel>();
+  {
+    sycl::queue Queue{Dev};
+    const sycl::context Ctx = Queue.get_context();
+    auto KernelBundle = sycl::get_kernel_bundle<sycl::bundle_state::input>(
+        Ctx, {Dev}, {KernelID_1});
+    const int globalWIs{512};
+    try {
+      auto event = Queue.submit([&](sycl::handler &cgh) {
+        cgh.parallel_for<InfoTestKernel>(globalWIs, [=](sycl::id<1> idx) {});
+      });
+      event.wait();
+      auto submit_time = event.get_profiling_info<
+          sycl::info::event_profiling::command_submit>();
+      (void)submit_time;
+    } catch (sycl::exception const &e) {
+      std::cerr << e.what() << std::endl;
+      EXPECT_STREQ(e.what(), "get_profiling_info() can't be used without set "
+                             "'enable_profiling' queue property");
+    }
+  }
+  {
+    sycl::queue Queue{Dev};
+    const sycl::context Ctx = Queue.get_context();
+    auto KernelBundle = sycl::get_kernel_bundle<sycl::bundle_state::input>(
+        Ctx, {Dev}, {KernelID_1});
+    const int globalWIs{512};
+    try {
+      auto event = Queue.submit([&](sycl::handler &cgh) {
+        cgh.parallel_for<InfoTestKernel>(globalWIs, [=](sycl::id<1> idx) {});
+      });
+      event.wait();
+      auto start_time =
+          event
+              .get_profiling_info<sycl::info::event_profiling::command_start>();
+      (void)start_time;
+    } catch (sycl::exception const &e) {
+      std::cerr << e.what() << std::endl;
+      EXPECT_STREQ(e.what(), "get_profiling_info() can't be used without set "
+                             "'enable_profiling' queue property");
+    }
+  }
+  {
+    sycl::queue Queue{Dev};
+    const sycl::context Ctx = Queue.get_context();
+    auto KernelBundle = sycl::get_kernel_bundle<sycl::bundle_state::input>(
+        Ctx, {Dev}, {KernelID_1});
+    const int globalWIs{512};
+    try {
+      auto event = Queue.submit([&](sycl::handler &cgh) {
+        cgh.parallel_for<InfoTestKernel>(globalWIs, [=](sycl::id<1> idx) {});
+      });
+      event.wait();
+      auto end_time =
+          event.get_profiling_info<sycl::info::event_profiling::command_end>();
+      (void)end_time;
+    } catch (sycl::exception const &e) {
+      EXPECT_STREQ(e.what(), "get_profiling_info() can't be used without set "
+                             "'enable_profiling' queue property");
+    }
+  }
```

Comment:
Would it make sense to add a couple of test cases where a queue goes out of scope before `get_profiling_info` is called on the event resulting from a `submit` on the now-dead queue?

Reply:
I think that would be good to check it. I'll implement your suggestion.